### PR TITLE
Add a built-in "print absolutely anything" function

### PR DESCRIPTION
**Pull request by dougalm**
The goal is to make this the default printing path and remove Interepreter/Serialize to unblock #1188. It'll also let us debug-print any dex value at runtime. There's still work to do in adding all the cases but I think it's worth merging this for now.

### CODE REVIEW

File: src/lib/Transpose.hs [hunk 1]
```diff
@@ -256,6 +256,8 @@ transposeMiscOp op _ = case op of
   BitcastOp    _ _      -> notImplemented
   UnsafeCoerce _ _      -> notImplemented
   GarbageVal _          -> notImplemented
+  ShowAny _ -> error "Shouldn't have ShowAny in simplified IR"
+  ShowScalar _ -> error "Shouldn't have ShowScalar in simplified IR"
```
Comment (axch) [inline src/lib/Transpose.hs:260]: We can't use IR tags to prove this doesn't happen?  Seems like an obvious use case for them...

**Review by axch — APPROVED** [review]
So, is generating code to print every possible type of value, inline, going to lead to bloated IRs and compile times?

Though, I don't suppose it would be any better to implement printing in Dex with classes, because I suppose it would amount to more or less the same thing.  Though if we marked the show function noinline, at least there would only be one copy of it per type, instead of per occurrence.

**Comment by dougalm** [comment]
> So, is generating code to print every possible type of value, inline, going to lead to bloated IRs and compile times?

We'll see. My hope is that the sorts of values we like to print have relatively small types. If it's an issue, the next step would be to add `@noinline` as you say.

File: src/lib/Transpose.hs (hunk 1 above)
Comment (dougalm) [reply src/lib/Transpose.hs:260]: Yes, absolutely. That will happen as part of #1188 .

**Pull request merged**